python/pyiceberg/catalog/glue.py

```diff
@@ -56,124 +64,109 @@
-        rename_table_input[PROP_GLUE_TABLE_DESCRIPTION] = table_description
+    rename_table_input["TableType"] = glue_table["TableType"]
+    if "Owner" in glue_table:
+        rename_table_input["Owner"] = glue_table["Owner"]
+
+    if "Parameters" in glue_table:
+        rename_table_input["Parameters"] = glue_table["Parameters"]
+
+    if "StorageDescriptor" in glue_table:
+        # It turns out the output of StorageDescriptor is not the same as the input type
+        # because the Column can have a different type, but for now it seems to work, so
+        # silence the type error.
+        rename_table_input["StorageDescriptor"] = cast(StorageDescriptorTypeDef, glue_table["StorageDescriptor"])
+
+    if "Description" in glue_table:
+        rename_table_input["Description"] = glue_table["Description"]
+
     return rename_table_input
 
 
-def _construct_database_input(database_name: str, properties: Properties) -> Dict[str, Any]:
-    database_input: Dict[str, Any] = {PROP_GLUE_DATABASE_NAME: database_name}
+def _construct_database_input(database_name: str, properties: Properties) -> DatabaseInputTypeDef:
+    database_input: DatabaseInputTypeDef = {"Name": database_name}
     parameters = {}
     for k, v in properties.items():
-        if k == GLUE_DESCRIPTION_KEY:
-            database_input[PROP_GLUE_DATABASE_DESCRIPTION] = v
+        if k == "Description":
+            database_input["Description"] = v
         elif k == LOCATION:
-            database_input[PROP_GLUE_DATABASE_LOCATION] = v
+            database_input["LocationUri"] = v
         else:
             parameters[k] = v
-    database_input[PROP_GLUE_DATABASE_PARAMETERS] = parameters
+    database_input["Parameters"] = parameters
     return database_input
 
 
 class GlueCatalog(Catalog):
-    def __init__(self, name: str, **properties: str):
+    def __init__(self, name: str, **properties: Any):
         super().__init__(name, **properties)
 
-        session_config = {k: v for k, v in properties.items() if k in BOTO_SESSION_CONFIG_KEYS}
-        session = boto3.Session(**session_config)
-        self.glue = session.client(GLUE_CLIENT)
+        session = boto3.Session(
+            profile_name=properties.get("profile_name"),
+            region_name=properties.get("region_name"),
+            botocore_session=properties.get("botocore_session"),
+            aws_access_key_id=properties.get("aws_access_key_id"),
+            aws_secret_access_key=properties.get("aws_secret_access_key"),
+            aws_session_token=properties.get("aws_session_token"),
+        )
+        self.glue: GlueClient = session.client("glue")
+
+    def _convert_glue_to_iceberg(self, glue_table: TableTypeDef) -> Table:
+        properties: Properties = glue_table["Parameters"]
 
-    def _convert_glue_to_iceberg(self, glue_table: Dict[str, Any]) -> Table:
-        properties: Properties = glue_table.get(PROP_GLUE_TABLE_PARAMETERS, {})
+        database_name = glue_table["DatabaseName"]
```

Comment:
Do we want to add null check for "Parameters" and "DatabaseName" as they are specified as `NotRequired` according to https://youtype.github.io/boto3_stubs_docs/mypy_boto3_glue/type_defs/#tabletypedef ?

Reply:
yes, I've added assertions for those properties.